library/libobject.mli

```diff
@@ -63,26 +64,26 @@ open Mod_subst
 
 *)
 
-type 'a substitutivity =
-    Dispose | Substitute of 'a | Keep of 'a | Anticipate of 'a
+type substitutivity = Dispose | Substitute | Keep | Anticipate
 
 (** Both names are passed to objects: a "semantic" [kernel_name], which
    can be substituted and a "syntactic" [full_path] which can be printed
 *)
 
-type object_name = full_path * Names.KerName.t
+type object_name = Libnames.full_path * KerName.t
 
 type open_filter
 
-type 'a object_declaration = {
+type ('a,'b) object_declaration = {
```

Comment:
Can't we instead get a GADT statically classifying the `object_name` type into three potential cases: unit, object_name and object_prefix? That would reduce the amount of cruft.

Reply:
>That would reduce the amount of cruft

doubt